### ISSUE TITLE
Question about Muon Optimizer Hyperparameter Settings

### ISSUE THREAD

**Issue by jie040109** [MoonshotAI/Moonlight#24]
Dear authors,
In the paper, you mentioned: “For Muon, as discussed in Sec 2.2, since we matched Muon’s update RMS to AdamW, we directly reused the hyper-parameters that are optimal for the AdamW baseline.”
Does this mean that when using Muon to train a new model, I can directly adopt the best hyperparameter settings previously used with AdamW on the same model? If not, could you kindly provide the recommended Muon hyperparameters for popular models such as GPT-2 and the LLaMA-2 series? Thank you very much!

**Comment by toothacher17** [comment]
hi, @jie040109 

> I can directly adopt the best hyperparameter settings previously used with AdamW on the same model? 
1. For the above question, the answer is yes. Our proposed method is to transfer the hyperparams for AdamW to Muon. 

2. Besides the overall principle of reusing AdamW's hyperparams, you still need to adjust the update rms as mentioned in the paper. This is to control the update rms to be the same as AdamW, so you can reuse AdamW's hyperparams.

The description of the method in the paper (where $\eta$ is the learning and $\lambda$ is the weight decay ratio used by AdamW):
![Image](https://github.com/user-attachments/assets/cc62f760-5471-420a-8029-3b17a2e32794)

An example code:
https://github.com/MoonshotAI/Moonlight/blob/master/examples/toy_train.py#L142
